File: backend/domain/expense.go
package domain

import "time"

type Expense struct {
	ID          string         `json:"id"`
	GroupID     string         `json:"group_id"`
	PaidBy      string         `json:"paid_by"`
	PaidByName  string         `json:"paid_by_name,omitempty"`
	Amount      float64        `json:"amount"`
	Description string         `json:"description,omitempty"`
	ReceiptURL  string         `json:"receipt_url,omitempty"`
	ExpenseDate string         `json:"expense_date"`
	CreatedAt   time.Time      `json:"created_at"`
	Splits      []ExpenseSplit `json:"splits,omitempty"`
}

type ExpenseSplit struct {
	ExpenseID  string  `json:"expense_id,omitempty"`
	UserID     string  `json:"user_id"`
	AmountOwed float64 `json:"amount_owed"`
}

type Balance struct {
	UserID  string  `json:"user_id"`
	Name    string  `json:"name"`
	Balance float64 `json:"balance"`
}

type GroupBalance struct {
	GroupID   string  `json:"group_id"`
	GroupName string  `json:"group_name"`
	Balance   float64 `json:"balance"`
}

type PersonBalance struct {
	UserID   string  `json:"user_id"`
	UserName string  `json:"user_name"`
	Balance  float64 `json:"balance"`
}

type Settlement struct {
	ID        string    `json:"id"`
	PayerID   string    `json:"payer_id"`
	PayeeID   string    `json:"payee_id"`
	Amount    float64   `json:"amount"`
	CreatedAt time.Time `json:"created_at"`
}

type OCRResult struct {
	Amount      float64 `json:"amount"`
	Description string  `json:"description"`
	Date        string  `json:"date"`
	RawText     string  `json:"raw_text"`
}

File: app/lib/domain/expense.dart
class Expense {
  final String id;
  final String groupId;
  final String paidBy;
  final String paidByName;
  final double amount;
  final String description;
  final String expenseDate;

  Expense({
    required this.id,
    required this.groupId,
    required this.paidBy,
    this.paidByName = '',
    required this.amount,
    required this.description,
    required this.expenseDate,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'] as String,
      groupId: json['group_id'] as String,
      paidBy: json['paid_by'] as String,
      paidByName: json['paid_by_name'] as String? ?? '',
      amount: (json['amount'] as num).toDouble(),
      description: json['description'] as String? ?? '',
      expenseDate: json['expense_date'] as String,
    );
  }
}

class Balance {
  final String userId;
  final String name;
  final double balance;

  Balance({required this.userId, required this.name, required this.balance});

  factory Balance.fromJson(Map<String, dynamic> json) {
    return Balance(
      userId: json['user_id'] as String,
      name: json['name'] as String? ?? json['user_name'] as String? ?? '',
      balance: (json['balance'] as num).toDouble(),
    );
  }
}

class OCRResult {
  final String description;
  final double amount;
  final String date;

  OCRResult({
    required this.description,
    required this.amount,
    required this.date,
  });

  factory OCRResult.fromJson(Map<String, dynamic> json) {
    return OCRResult(
      description: json['description'] as String? ?? '',
      amount: (json['amount'] as num?)?.toDouble() ?? 0.0,
      date: json['date'] as String? ?? '',
    );
  }
}
